ts/client/services/rpc.ts

```diff
@@ -20,12 +22,70 @@ export class RPC {
         fastify: FastifyInstance,
         transferPool?: ITransferPool
     ) {
+        fastify.register(cors, {
+            origin: (origin, cb) => {
+                if (/localhost/.test(origin)) {
+                    //  Request from localhost will pass
+                    cb(null, true);
+                    return;
+                }
+                // Generate an error on other origins, disabling access
+                cb(new Error("Not allowed"), false);
+            }
+        });
         fastify.get<{ Params: { stateID: number } }>(
             "/user/state/:stateID",
-            async function(request) {
-                const stateID = request.params.stateID;
-                const state = await l2Storage.state.get(stateID);
-                return state.toJSON();
+            async function(request, reply) {
+                try {
+                    const { stateID } = request.params;
+                    const state = await l2Storage.state.get(stateID);
+                    return state.toJSON();
+                } catch (error) {
+                    return reply.status(404).send({ error: "state not found" });
```

Comment:
`error.name === "NotFoundError"` check here for 404, otherwise should be a `HTTP 500`